### PR TITLE
chore: use new junit2otel step

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It uses the `junit2otel` new step in this pipeline, setting the `JUNIT_2_OTLP` env var at the top-level environment section of the pipeline.

Because we are not passing service name/version or trace name, the step will infer them from:

- REPO as service name and trace name
- service version will be calculated from:
    - env.CHANGE_ID on PRs
    - env.TAG_NAME on tags
    - env.BRANCH_NAME on branches
    - `unknown` in any other case

We may need to improve this logic to automatically detect the project type:
- is it a Maven project? read the version from the pom.xml
- ...

## Why is it important?
Let's start collecting metrics from test results!

### CODE REVIEW

**Review by v1v — APPROVED** [review]

**Comment by mdelapenya** [comment]
Need to figure it out why this error:

```
[2022-02-11T12:52:18.994Z] SERVICE_VERSION=1543
[2022-02-11T12:52:18.994Z] TEST_RESULTS_GLOB=src/github.com/elastic/apm-pipeline-library/target/surefire-reports/junit-*.xml,src/github.com/elastic/apm-pipeline-library/target/surefire-reports/TEST-*.xml
[2022-02-11T12:52:18.994Z] TRACE_NAME=apm-pipeline-library
[2022-02-11T12:52:18.994Z] Sending traces for src/github.com/elastic/apm-pipeline-library/target/surefire-reports/junit-*.xml file...
[2022-02-11T12:52:18.994Z] /var/lib/jenkins/workspace/apm-pipeline-library-mbp_PR-1543@tmp/durable-2e1da3e5/script.sh: line 35: OTEL_EXPORTER_OTLP_HEADERS: unbound variable
[2022-02-11T12:52:18.994Z] cat: 'src/github.com/elastic/apm-pipeline-library/target/surefire-reports/junit-*.xml': No such file or directory
script returned exit code 1
```

**Comment by mdelapenya** [comment]
Now it works, although the traces are not connected to the parent trace.

@v1v let's dig into this Monday 🙏

**Review by kuisathaverat — APPROVED** [review]

**Comment by mdelapenya** [comment]
I can see the `apm-junit` traces and spans for the distributed traces of the pipeline (see screenshots below), but what I do not see is the Git SCM attributes. I would need to do more testing for that, because the junit2otel CLI looks for the `.git` directory at the location of the tool, and as far as I can check I'm not mounting the current workspace to the Docker image for the CLI in the shell script call. Will send another commit to fix that

### Jaeger
Screenshot 2022-02-16 at 07 29 34

Screenshot 2022-02-16 at 07 27 59

### Elastic APM UI
Screenshot 2022-02-16 at 07 33 55

**Review by kuisathaverat — APPROVED** [review]

**Comment by mdelapenya** [comment]
Fixing the SCM path in https://github.com/mdelapenya/junit2otlp/pull/22

**Comment by mdelapenya** [comment]
Ok, cool, now I can see the SCM contributions, but still not seeing those related to a PR, like authors or modified files:
Screenshot 2022-02-16 at 11 53 20

@v1v @kuisathaverat do you think it's because of the git checkout we perform on Jenkins? I can see the branch name is `HEAD`

**Comment by mdelapenya** [comment]
I found the issue, and it's related to this: because we are running the CLI on Jenkins, the SCM provider of the CLI is not detecting it as a change request, because it's only setting that for Github actions and Gitlab runners. Because we have not added the JUnit support to the OTEL Jenkins plugin yet, we are in the need to use the CLI on Jenkins builds. Therefore, we should update the CLI to detect if it's run on Jenkins to get all the information for the SCM

**Comment by mdelapenya** [comment]
@v1v @kuisathaverat I'd say this is ready to be merged. I can see the SCM attributes although I'm missing those related to the committers. If you think this is OK, we can merge, and resolve the scm issues in the CLI side, as we are consuming `latest` Docker image, so any change/improvement will be resolved at that level

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:18]: To be reverted after it gets merged?

Comment (mdelapenya) [reply .ci/Jenkinsfile:18]: How do you usually do with kind of use cases, where we need a custom branch to verify the new behaviours?

Comment (v1v) [reply .ci/Jenkinsfile:18]: 1) PR from my fork (with the steps/changes but without changing the Jenkinsfile)
2) Branch called `test/....` in the upstream with the changes in the Jenkinsfile
3) If the branch works, then I merge my PR
4) Create a branch from 2) with the changes in the Jenkinsfile.

Another approach could be:
1) PR from my fork
2) Create pipeline in the local docker jenkins instance (and use those new steps)
3) Merge the PR
4) Create PR with the changes in the Jenkinsfile

Comment (mdelapenya) [reply .ci/Jenkinsfile:18]: Got it. I think this branch is following the first approach, so I can:
1. revert the usage of the new step in the Jenkinsfile
2. merge this PR
3. release the library
4. create a new PR with the changes in the Jenkinsfile

Does it sound good?

Comment (v1v) [reply .ci/Jenkinsfile:18]: > release the library

You don't need this, it uses `main` :)

But  I need to release a new version shortly, so I'll wait for you

Comment (mdelapenya) [reply .ci/Jenkinsfile:18]: Could doing it asap

Comment (mdelapenya) [reply .ci/Jenkinsfile]: @v1v I removed any change in the Jenkinsfile regarding the new step. I think this is ready to be merged

**Review by v1v — APPROVED** [review]

**Pull request merged**
